src/DynamoCore/Models/WorkspaceModel.cs

```diff
@@ -1445,8 +1489,8 @@ public void CreateModel(XmlElement modelData)
             else // Other node types.
             {
                 NodeModel nodeModel = NodeFactory.CreateNodeFromXml(modelData, SaveContext.Undo);
-                Nodes.Add(nodeModel);
-                RegisterNode(nodeModel);
```

Comment:
Sadly it isn't a no-op, it's really nasty pattern that the code had, where it needed to call this exactly once after modifications to the nodes to ensure coherence, which is really error prone. I've moved it into a post-operation call on the AddNode method.